pkg/result/processors/base_rule.go

```diff
@@ -26,14 +27,14 @@ func (r *baseRule) isEmpty() bool {
 	return r.text == nil && r.source == nil && r.path == nil && len(r.linters) == 0
 }
 
-func (r *baseRule) match(issue *result.Issue, lineCache *fsutils.LineCache, log logutils.Log) bool {
+func (r *baseRule) match(issue *result.Issue, pathPrefix string, lineCache *fsutils.LineCache, log logutils.Log) bool {
 	if r.isEmpty() {
 		return false
 	}
 	if r.text != nil && !r.text.MatchString(issue.Text) {
 		return false
 	}
-	if r.path != nil && !r.path.MatchString(issue.FilePath()) {
+	if r.path != nil && !r.path.MatchString(path.Join(pathPrefix, issue.FilePath())) {
```

Comment:
If `pathPrefix` is empty, `pathPrefix` will ignore it. But the result is "cleaned" - I am not sure whether that is potentially harmful (performance, Windows separator?).

If "path prefix" was an attribute in `lineCache`, then a `LineCache.WithPathPrefix(p string)` method could be added to do the extension of the path in a consistent way.

Reply:
`path.Join` is correct. The path prefixer does the same. To handle Windows, the path regex gets mangled to match the separator used by the platform.